pkg/detectors/gcp/gcp.go

```diff
@@ -90,41 +103,144 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 
 		credBytes, _ := json.Marshal(creds)
```

Comment:
Do we need to actually capture the whole json blob and marshal the json? One issue with GCP keys is they're so big sometimes they get cut off in scanning windows. Maybe now that we've updated verifier logic, we could get away with just a capture group around the private key, and a capture group around the x509 URL

Reply:
Technically all that's required is the `private_key` and `client_email` (`name@project_id.iam.gserviceaccount.com`). The X509 URL is just `https://www.googleapis.com/robot/v1/metadata/x509/$client_email`

Reply:
Either the x509 url or the client_email work as capture groups.